src/common/utils/get_resolved_document_by_id.py

```diff
@@ -23,27 +17,21 @@ def resolve_reference_list(
 
     if isinstance(value_sample, list):  # Call recursively for nested lists
         return [
-            resolve_reference_list(
-                value, document_repository, get_data_source, current_id, depth, depth_count, resolve_links
-            )
+            resolve_reference_list(value, document_repository, get_data_source, current_id, depth, depth_count)
             for value in values
         ]
 
     if is_reference(value_sample):
-        if resolve_links or not is_link(value_sample):
+        if depth_count <= depth:
```

Comment:
This one is always true. Checking depth vs depth_count is done by resolve_references_in_entity, and since that check must pass before this function can be called, the check is always true.